assets/src/stories-editor/blocks/amp-story-page-attachment/edit.js

```diff
@@ -0,0 +1,301 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, RawHTML } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { InspectorControls, RichText } from '@wordpress/block-editor';
+import { select } from '@wordpress/data';
+import {
+	SelectControl,
+	PanelBody,
+	Button,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './edit.css';
+import { getBackgroundColorWithOpacity } from '../../../common/helpers';
+import { PostSelector } from '../../components';
+
+class PageAttachmentEdit extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			selectedPost: null,
+			isOpen: false,
+			searchValue: '',
+			failedToFetch: false,
+		};
+
+		this.toggleAttachment = this.toggleAttachment.bind( this );
+		this.removePost = this.removePost.bind( this );
+	}
+
+	componentDidMount() {
+		this.fetchSelectedPost();
+	}
+
+	componentWillUnmount() {
+		this.isStillMounted = false;
+	}
+
+	componentDidUpdate( prevProps ) {
```

Comment:
Initially when the block got unselected then the attachment was also closed, however, this caused the attachment to lose focus sometimes in some weird moments and it was quite disturbing. Now the attachment stays open until it's closed from `X` even if not selected.

Not sure what would be the best approach here. You can test the previous version out when adding this:
```
                if ( ! attributes.isSelected && prevProps.isSelected ) {
			this.toggleAttachment( false );
		}
```

Thoughts?

Reply:
Makes sense to me. There are quite a few ways to accidentally unselect a block.